plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestHiveMetastoreMetadataQueriesAccessOperations.java

```diff
@@ -0,0 +1,511 @@
+                ImmutableMultiset.builder()
+                        .add(GET_ALL_TABLES_FROM_DATABASE)
+                        .add(GET_ALL_VIEWS_FROM_DATABASE)
+                        .addCopies(GET_TABLE, TEST_TABLES_IN_SCHEMA_COUNT)
+                        .build());
+        assertMetastoreInvocations("SELECT * FROM system.jdbc.columns WHERE table_schem = 'test_schema_0'",
+                ImmutableMultiset.builder()
+                        .add(GET_ALL_TABLES_FROM_DATABASE)
+                        .add(GET_ALL_VIEWS_FROM_DATABASE)
+                        .addCopies(GET_TABLE, TEST_TABLES_IN_SCHEMA_COUNT)
+                        .build());
+        assertMetastoreInvocations("SELECT * FROM system.jdbc.columns WHERE table_schem LIKE 'test\\_schema\\_0' ESCAPE '\\'",
+                ImmutableMultiset.builder()
+                        .add(GET_ALL_TABLES_FROM_DATABASE)
+                        .add(GET_ALL_VIEWS_FROM_DATABASE)
+                        .addCopies(GET_TABLE, TEST_TABLES_IN_SCHEMA_COUNT)
+                        .build());
+        assertMetastoreInvocations("SELECT * FROM system.jdbc.columns WHERE table_schem LIKE 'test_schema_0' ESCAPE '\\'",
+                ImmutableMultiset.builder()
+                        .add(GET_ALL_DATABASES)
+                        .add(GET_ALL_TABLES_FROM_DATABASE)
+                        .add(GET_ALL_VIEWS_FROM_DATABASE)
+                        .addCopies(GET_TABLE, TEST_TABLES_IN_SCHEMA_COUNT)
+                        .build());
+    }
+
+    @Test
+    public void testSelectColumnsWithLikeOverSchema()
+    {
+        assertMetastoreInvocations("SELECT * FROM information_schema.columns WHERE table_schema LIKE 'test%'",
+                ImmutableMultiset.builder()
+                        .add(GET_ALL_DATABASES)
+                        .addCopies(GET_ALL_TABLES_FROM_DATABASE, TEST_SCHEMAS_COUNT)
+                        .addCopies(GET_ALL_VIEWS_FROM_DATABASE, TEST_SCHEMAS_COUNT)
+                        .addCopies(GET_TABLE, TEST_ALL_TABLES_COUNT)
+                        .build());
+        assertMetastoreInvocations("SELECT * FROM system.jdbc.columns WHERE table_schem LIKE 'test%'",
+                ImmutableMultiset.builder()
+                        .add(GET_ALL_DATABASES)
+                        .addCopies(GET_ALL_TABLES_FROM_DATABASE, TEST_SCHEMAS_COUNT)
+                        .addCopies(GET_TABLE, TEST_ALL_TABLES_COUNT)
+                        .build());
+    }
+
+    @Test
+    public void testSelectColumnsFilterByTableName()
+    {
+        metastore.resetCounters();
+        computeActual("SELECT * FROM information_schema.columns WHERE table_name = 'test_table_0'");
+        Multiset<Methods> invocations = metastore.getMethodInvocations();
+
+        assertThat(invocations.count(GET_TABLE)).as("GET_TABLE invocations")
+                // some lengthy explanatory comment why variable count
```

Comment:
"some lengthy explanatory comment why variable count" from https://github.com/trinodb/trino/pull/17231#discussion_r1177865238 wasn't supposed to be treated literally, it was a placeholder :)